model/task/task.go

```diff
@@ -2797,6 +2757,38 @@ func (t *Task) SetResetWhenFinished() error {
 	)
 }
 
+// SetResetWhenFinishedWithInc requests that a task (that was marked to
+// automatically reset when finished via the agent status server) reset itself
+// when finished. It will also increment the number of automatic resets the task
+// has performed.
+func (t *Task) SetResetWhenFinishedWithInc() error {
+	if t.ResetWhenFinished {
+		return nil
+	}
+	err := UpdateOne(
+		bson.M{
+			IdKey:                 t.Id,
+			AbortedKey:            bson.M{"$ne": true},
+			IsAutomaticRestartKey: bson.M{"$ne": true},
+		},
+		bson.M{
+			"$set": bson.M{
+				ResetWhenFinishedKey:  true,
+				IsAutomaticRestartKey: true,
+			},
+			"$inc": bson.M{
+				NumAutomaticRestartsKey: 1,
+			},
+		},
+	)
+	if err != nil {
+		return err
+	}
+	t.ResetWhenFinished = true
```

Comment:
I think we should just drop these updates, since the inc might be wrong unless we include it in the query (which we might want to do if this can be set in more than one command, since it could race), and its inconsistent with the other retry functions (and the fields aren't used)

Reply:
Fine by me, removed